lib/client/client.go

```diff
@@ -402,17 +402,49 @@ func (proxy *ProxyClient) dialAuthServer(ctx context.Context, clusterName string
 	), nil
 }
 
+// NodeAddr is a full node address
+type NodeAddr struct {
+	// Addr is an address to dial
+	Addr string
+	// Namespace is the node namespace
+	Namespace string
+	// Cluster is the name of the target cluster
+	Cluster string
+}
+
+// String returns a user-friendly name
+func (n NodeAddr) String() string {
+	parts := []string{nodeName(n.Addr)}
+	if n.Cluster != "" {
+		parts = append(parts, "on cluster", n.Cluster)
+	}
+	return strings.Join(parts, " ")
+}
+
+// ProxyFormat returns the address in the format
+// used by the proxy subsystem
+func (n *NodeAddr) ProxyFormat() string {
```

Comment:
Not familiar with how Teleport handles subsystem naming, but the output of this function may be either only the namespace or only the address if one of the two is missing.  Is this ambiguity OK?

Reply:
address is always present, so this use-case is unlikely, I guess we can make a constructor that can enforce the invariant, but I think it's ok to fail here miserably if we miss the address elsewhere